src/StaticAssets/src/Development/StaticAssetDescriptorExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Microsoft.AspNetCore.StaticAssets;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Builder;
+
+internal static class StaticAssetDescriptorExtensions
+{
+    internal static long GetContentLength(this StaticAssetDescriptor descriptor)
+    {
+        foreach (var header in descriptor.ResponseHeaders)
+        {
+            if (header.Name == "Content-Length")
```

Comment:
Is it okay that we're not doing a case insensitive comparison here (and elsewhere in this PR when checking header names)?

Reply:
It should be fine since we generate this data, however, for resiliency I've changed it to use `string.Equals(... OrdinalIgnoreCase)`